File: glass/src/lib/native/cpp/hardware/Accelerometer.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "glass/hardware/Accelerometer.h"

#include "glass/DataSource.h"
#include "glass/other/DeviceTree.h"

using namespace glass;

void glass::DisplayAccelerometerDevice(AccelerometerModel* model) {
  if (!model->Exists()) {
    return;
  }
  if (BeginDevice("BuiltInAccel")) {
    // Range
    {
      int value = model->GetRange();
      static const char* rangeOptions[] = {"2G", "4G", "8G"};
      DeviceEnum("Range", true, &value, rangeOptions, 3);
    }

    // X Accel
    if (auto xData = model->GetXData()) {
      double value = xData->GetValue();
      if (DeviceDouble("X Accel", false, &value, xData)) {
        model->SetX(value);
      }
    }

    // Y Accel
    if (auto yData = model->GetYData()) {
      double value = yData->GetValue();
      if (DeviceDouble("Y Accel", false, &value, yData)) {
        model->SetY(value);
      }
    }

    // Z Accel
    if (auto zData = model->GetZData()) {
      double value = zData->GetValue();
      if (DeviceDouble("Z Accel", false, &value, zData)) {
        model->SetZ(value);
      }
    }

    EndDevice();
  }
}

File: wpilibj/src/main/java/edu/wpi/first/wpilibj/interfaces/Accelerometer.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package edu.wpi.first.wpilibj.interfaces;

/**
 * Interface for 3-axis accelerometers.
 *
 * @deprecated This interface is being removed with no replacement.
 */
@Deprecated(since = "2024", forRemoval = true)
public interface Accelerometer {
  /** Accelerometer range. */
  enum Range {
    /** 2 Gs max. */
    k2G,
    /** 4 Gs max. */
    k4G,
    /** 8 Gs max. */
    k8G,
    /** 16 Gs max. */
    k16G
  }

  /**
   * Common interface for setting the measuring range of an accelerometer.
   *
   * @param range The maximum acceleration, positive or negative, that the accelerometer will
   *     measure. Not all accelerometers support all ranges.
   */
  void setRange(Range range);

  /**
   * Common interface for getting the x-axis acceleration.
   *
   * @return The acceleration along the x-axis in g-forces
   */
  double getX();

  /**
   * Common interface for getting the y-axis acceleration.
   *
   * @return The acceleration along the y-axis in g-forces
   */
  double getY();

  /**
   * Common interface for getting the z axis acceleration.
   *
   * @return The acceleration along the z axis in g-forces
   */
  double getZ();
}
